distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Type_Refinements/Table_Refinements.enso

```diff
@@ -0,0 +1,38 @@
+private
+
+from Standard.Base import all
+
+import project.Column.Column
+import project.Table.Table
+import project.Value_Type.Value_Type
+from project.Internal.Type_Refinements.Single_Column_Table_Conversions import all
+from project.Internal.Type_Refinements.Column_Refinements import is_single_value
+
+refine_table (table : Table) =
+    if is_single_column table . not then table else
+        column = table.at 0
+        # This should be consistent with `Column_Refinements.refine_column` - the code needs to be copied because we need to spell out all the types.
```

Comment:
Probably one we need to talk over with Jaroslav to see if we can do anything better longer term.